src/Lecture2.hs

```diff
@@ -159,8 +180,43 @@ data Knight = Knight
     , knightAttack    :: Int
     , knightEndurance :: Int
     }
-
-dragonFight = error "TODO"
+data Color
+  = Red
+  | Green
+  | Black
+
+
+data Treasure
+  = Gold
+  | GoldPlusTreasure
```

Comment:
I clarified the task a bit to cover cases so dragons would determine the treasure itself, not just "yes" or "no". It will make the implementation a bit more challenging 😅 

But, besides that, the current implementation covers the requirements nicely 👍🏻